esrally/driver/runner.py

```diff
@@ -643,11 +645,7 @@ async def __call__(self, es, params):
         except elasticsearch.TransportError as e:
             # this is caused by older versions of Elasticsearch (< 2.1), fall back to optimize
             if e.status_code == 400:
-                if max_num_segments:
-                    await es.transport.perform_request("POST", f"/_optimize?max_num_segments={max_num_segments}",
-                                                       timeout=request_timeout)
-                else:
-                    await es.transport.perform_request("POST", "/_optimize", timeout=request_timeout)
+                await es.transport.perform_request(method="POST", url="/_optimize", params=merge_params)
```

Comment:
`transport.perform_request()` expects the `timeout` parameter in the `params` arg.  Throws
```
TypeError: perform_request() got an unexpected keyword argument 'timeout'
```

Reply:
it looks like you removed the logic here about `max_num_segments`. Good catch about the timeout though, lets definitely fix that here once you revert the code change that deleted the `if` stmt

Reply:
max_num_segments exists in the merge_params and gets applied to the query string by the perform_request method